File: macos/Scanner/Sources/Scanner/Detectors/ClaudeCliDetector.swift
public struct ClaudeCliDetector: Detector {
    public let id = "claude-cli"
    public let displayName = "Claude CLI"

    public init() {}

    public func collectFacts(fs: any FilesystemFacade) throws -> ScanFacts {
        // Global config: ~/.claude.json (macOS/Linux) or %USERPROFILE%\.claude.json (Windows)
        let configPath = homePath(fs, ".claude.json")
        var facts = ScanFacts()
        appendAppInstallationFact(
            &facts,
            fs: fs,
            appId: id,
            candidatePaths: [homePath(fs, ".claude")]
        )

        if let text = try readIfPresent(fs, path: configPath) {
            facts.mcpServers.append(contentsOf: try FactParsers.parseMcpServersJson(text, appId: id, configPath: configPath))
            appendConfigFileFact(&facts, fs: fs, appId: id, path: configPath)
        }

        return facts
    }
}

File: windows/Scanner/Detectors/ClaudeCliDetector.cs
using AIExposureScanner.Scanner.Filesystem;
using AIExposureScanner.Scanner.Models;
using AIExposureScanner.Scanner.Parsing;

namespace AIExposureScanner.Scanner.Detectors;

public sealed class ClaudeCliDetector : IDetector
{
    public string Id => "claude-cli";
    public string DisplayName => "Claude CLI";

    public ScanFacts CollectFacts(IFilesystemFacade fs)
    {
        // Global config: ~/.claude.json (macOS/Linux) or %USERPROFILE%\.claude.json (Windows)
        var configPath = DetectorSupport.HomePath(fs, ".claude.json");

        var facts = new ScanFacts();
        DetectorSupport.AppendAppInstallationFact(
            facts,
            fs,
            Id,
            [DetectorSupport.HomePath(fs, ".claude")]
        );

        var text = DetectorSupport.ReadIfPresent(fs, configPath);
        if (text is not null)
        {
            facts.McpServers.AddRange(FactParsers.ParseMcpServersJson(text, Id, configPath));
            DetectorSupport.AppendConfigFileFact(facts, fs, Id, configPath);
        }

        return facts;
    }
}
